tests/integration/test_factory.py

```diff
@@ -1,48 +1,62 @@
 import os
-
-from requre import RequreTestCase
-from requre.storage import PersistentObjectStorage
-from requre.utils import StorageMode
+from requre.online_replacing import record_requests
+from requre.utils import get_datafile_filename
+import unittest
 
 from ogr import GithubService, PagureService, get_project, GitlabService
 from ogr.services.github import GithubProject
 from ogr.services.gitlab import GitlabProject
 from ogr.services.pagure import PagureProject
 
 
-class FactoryTests(RequreTestCase):
+class FactoryTests(unittest.TestCase):
     def setUp(self):
-        super().setUp()
-        print(PersistentObjectStorage().storage_file)
+        self._github_service = None
+        self._pagure_service = None
+        self._gitlab_service = None
         self.github_token = os.environ.get("GITHUB_TOKEN")
         self.pagure_token = os.environ.get("PAGURE_TOKEN")
         self.gitlab_token = os.environ.get("GITLAB_TOKEN") or "some_token"
-
-        if PersistentObjectStorage().mode == StorageMode.write:
-            if not self.github_token:
-                raise EnvironmentError(
-                    "You are in requre write mode, please set GITHUB_TOKEN env variables"
-                )
-            if not self.pagure_token:
-                raise EnvironmentError(
-                    "You are in requre write mode, please set PAGURE_TOKEN env variables"
-                )
-            if not os.environ.get("GITLAB_TOKEN"):
+        if not os.path.exists(get_datafile_filename(obj=self)):
+            if (
+                not self.github_token
+                or not self.pagure_token
+                or not os.environ.get("GITLAB_TOKEN")
+            ):
                 raise EnvironmentError(
-                    "You are in requre write mode, please set GITLAB_TOKEN env variables"
+                    "You are in requre write mode, please set GITHUB_TOKEN PAGURE_TOKEN"
+                    " GITLAB_TOKEN env variables"
                 )
 
-        self.github_service = GithubService(token=self.github_token)
-        self.pagure_service = PagureService(token=self.pagure_token)
-        self.gitlab_service = GitlabService(
-            token=self.gitlab_token, instance_url="https://gitlab.com"
-        )
-        self.custom_instances = [
+    @property
```

Comment:
Is it regenerated for each test case or only once?

Reply:
for each test case , I've tried to test it, that I've regenerated all outputs and then try to run second test, and seems to work as expected.

Reply:
Theoretically also code like: 
```
    @pytest.fixture(autouse=True)
    @record_request
    def prepare_flag(self):
           yield
   ```
coud work as expected, to decorate each test method, as you've sent me link to https://stackoverflow.com/questions/50710028/how-to-use-pytest-fixtures-with-unittest-methods
but not sure if it is readable enough, and does not cause confusion,
I try to look at it and try to find the best way how to do it.

Reply:
yep, I've tested, the code like: 

```
def for_all_methods(decorator, methond_pattern="test.*"):
    def decorate(cls):
        for attr in cls.__dict__:
            if callable(getattr(cls, attr)) and re.match(methond_pattern, attr):
                setattr(cls, attr, decorator(getattr(cls, attr)))
        return cls
    return decorate

@for_all_methods(record_requests)
class TestClass(unittest.TestCase)
...
```
works as expected. I can incorporate it into `requre` as well, that's true that decoration every test function, especially inside more extensive testsuite like github, will contain lots of this same decorators. what do you think?

Reply:
Nice! Is it possible to have this in some supper class?

Reply:
> Nice! Is it possible to have this in some supper class?

@lachmanfrantisek not sure. therein text what you've linked is used metaclass, to override `__new__` constructor to apply these decorators, but from my perspective it will be more cryptic to use these metaclasses.

Reply:
@jscotka I would go with the class decorator and users can decide if they want to mark class or method.

What about this:

```python
def record_requests_for_all_methods(methond_pattern="*"):
    def decorate(cls):
        for attr in cls.__dict__:
            if callable(getattr(cls, attr)) and re.match(methond_pattern, attr):
                setattr(cls, attr, record_requests(getattr(cls, attr)))
        return cls
    return decorate
```

Reply:
moved generic code to https://github.com/packit/requre/pull/112 and then I'll use it here. Your suggestion will be useful just for `ogr` where is used just requests module to be decorated.